secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/sign/DefaultSapiJwsSigner.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.sign;
+
+import static org.forgerock.openig.secrets.SecretsProviderHeaplet.secretsProvider;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Objects;
+
+import org.forgerock.json.jose.builders.JwsHeaderBuilder;
+import org.forgerock.json.jose.builders.JwtBuilderFactory;
+import org.forgerock.json.jose.jws.JwsAlgorithm;
+import org.forgerock.json.jose.jws.SigningManager;
+import org.forgerock.json.jose.jws.handlers.SigningHandler;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.openig.heap.GenericHeaplet;
+import org.forgerock.openig.heap.HeapException;
+import org.forgerock.secrets.Purpose;
+import org.forgerock.secrets.SecretsProvider;
+import org.forgerock.secrets.keys.SigningKey;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.Promises;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation of {@link SapiJwsSigner}
+ * <p>
+ * A default JWS signer. <br/>
+ * Provides signing functionality for use by the filters and/or the IG scripts<br/>
+ * @see DefaultSapiJwsSigner.Heaplet
+ */
+public class DefaultSapiJwsSigner implements SapiJwsSigner {
```

Comment:
I don't like the Default part of this name. You should describe what thsi signer does. What is default about it? It's only default because there aren't any others! 

For example you might call it CompactSerializationJwsSigner (why put SAPI in there?)

Reply:
You are right, It has been implemented to sign events, maybe the Default part of the name isn't accurate.
What about change it to `EventsJwsSigner`?

If the customer can implement his own signer to address his own cases (different algorithm, kid, secret provider etc), we will provide the proper documentation to implement a custom signer and configure it in the IG heap.

We do not intend to implement a generic signer to cover all possible cases.

I have use the Sapi prefix in the name because I have remember a conversation that we had about to use that prefix in IG extensions classes, we have few of classes currently where we use that prefix.